src/java/org/apache/cassandra/io/sstable/format/bti/BtiFormat.java

```diff
@@ -0,0 +1,490 @@
+
+            isLatestVersion = version.compareTo(current_version) == 0;
+            hasOldBfFormat = version.compareTo("b") < 0;
+            hasAccurateMinMax = version.compareTo("ac") >= 0;
+            hasOriginatingHostId = version.matches("(a[d-z])|(b[b-z])") || version.compareTo("ca") >= 0;
+            hasImprovedMinMax = version.compareTo("ba") >= 0;
+            hasKeyRange = version.compareTo("da") >= 0;
+            hasPartitionLevelDeletionsPresenceMarker = version.compareTo("ba") >= 0;
+            correspondingMessagingVersion = MessagingService.VERSION_40;
+        }
+
+        // this is for the ab version which was used in the LABS, and then has been renamed to ba
+        private static String mapAb(String version)
+        {
+            return "ab".equals(version) ? "ba" : version;
+        }
+
+        @Override
+        public boolean isLatestVersion()
+        {
+            return isLatestVersion;
+        }
+
+        // this field is not present in DSE
+        @Override
+        public int correspondingMessagingVersion()
+        {
+            return correspondingMessagingVersion;
+        }
+
+        @Override
+        public boolean hasCommitLogLowerBound()
+        {
+            return true;
+        }
+
+        @Override
+        public boolean hasCommitLogIntervals()
+        {
+            return true;
+        }
+
+        @Override
+        public boolean hasMaxCompressedLength()
+        {
+            return true;
+        }
+
+        @Override
+        public boolean hasPendingRepair()
+        {
+            return true;
+        }
+
+        // this field is not present in DSE
+        @Override
+        public boolean hasIsTransient()
+        {
+            return version.compareTo("ca") >= 0;
+        }
+
+        @Override
+        public boolean hasMetadataChecksum()
+        {
+            return true;
+        }
+
+        @Override
+        public boolean hasOldBfFormat()
+        {
+            return hasOldBfFormat;
+        }
+
+        @Override
+        public boolean hasAccurateMinMax()
+        {
+            return hasAccurateMinMax;
+        }
+
+        public boolean hasLegacyMinMax()
+        {
+            return false;
+        }
+
+        @Override
+        public boolean hasOriginatingHostId()
+        {
+            return hasOriginatingHostId;
+        }
+
+        @Override
+        public boolean hasImprovedMinMax() {
+            return hasImprovedMinMax;
+        }
+
+        @Override
+        public boolean hasPartitionLevelDeletionsPresenceMarker()
+        {
+            return hasPartitionLevelDeletionsPresenceMarker;
+        }
+
+        @Override
+        public boolean hasKeyRange()
+        {
+            return hasKeyRange;
+        }
+
+        @Override
+        public boolean isCompatible()
+        {
+            return version.compareTo(earliest_supported_version) >= 0 && version.charAt(0) <= current_version.charAt(0);
```

Comment:
I think this support version should have a ut case when the current_version and earliest_supported_version change to apache cassandra support version format.

Reply:
I don't understand the comment.

This follows the pattern in existing format handling, where we support older and current major versions (i.e. `charAt(0)`), and we also accept future minor version changes for all supported majors.

Reply:
ok